docs/false_positives.md

```diff
@@ -235,6 +236,84 @@ the analyzer better. Note that the compiler might generate the same code
 for the two implementations, so it might make sense to use only the more
 obvious one.
 
+## <a name="defensive-checks"></a> Defensive checks
+
+This section is an odd one because we describe a technique to reduce false
+negatives rather than false positives. Nevertheless this is a useful tool to
+know and we have yet to find a better place to put this.
+
+The preconditions of a function should not be violated by the callers.
+For the working example we will look at an `strlen` implementation.
+It is illegal to call `strlen` with a null pointer.
+
+```cpp
+int strlen(const char *c) {
+  int res = 0;
+  for(; *c != 0; ++c) // Warn! Null pointer dereference.
+    ++res;
+  return res;
+}
+
+int main() {
+  return strlen(0);
+}
+```
+
+The analyzer will be able to find the null dereference in the code above.
+In some cases, however, the author of the functions adds defensive checks to
+avoid crashes when some clients do not respect the precondition.
+
+```cpp
+int strlen(const char *c) {
+  if (c == 0)
+    return 0; // Or might throw exception in case of C++.
+  int res = 0;
+  for(; *c != 0; ++c) // No warning.
+    ++res;
+  return res;
+}
+
+int main() {
+  return strlen(0);
+}
+```
+
+In the code above the analyzer will not be able to find any errors since it has
+no way to tell whether the defensive checks are due to contract violations or
+they are part of the defined behavior.
+
+In order to reduce false negatives due to those safety checks we have several
```

Comment:
I would elaborate more on how this particular case results in a false negative. (It might be me being out of the loop as I haven't looked at code for weeks now, but can't seem to figure out at first or second read.)

Reply:
The paragraph above this one supposed to explain this.